### PR TITLE
[SPARK-50494][INFRA] Add a separate docker file for python 3.10 daily build

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Add a separate docker file for python 3.10 daily build

### Why are the changes needed?
To isolate the envs

### Does this PR introduce _any_ user-facing change?
no, infra-only

### How was this patch tested?
test PR builder with envs:

```
default: '{"PYSPARK_IMAGE_TO_TEST": "python-310", "PYTHON_TO_TEST": "python3.10"}'
```

https://github.com/zhengruifeng/spark/runs/33973040152

### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

**Comment by zhengruifeng** [comment]
step `Free up disk space`

before:
```
Filesystem      Size  Used Avail Use% Mounted on
overlay          73G   63G   11G  87% /
tmpfs            64M     0   64M   0% /dev
shm              64M     0   64M   0% /dev/shm
/dev/root        73G   63G   11G  87% /__w
tmpfs           3.2G  1.2M  3.2G   1% /run/docker.sock
tmpfs           7.9G     0  7.9G   0% /proc/acpi
tmpfs           7.9G     0  7.9G   0% /proc/scsi
tmpfs           7.9G     0  7.9G   0% /sys/firmware
Removing large packages
Filesystem      Size  Used Avail Use% Mounted on
overlay          73G   57G   17G  78% /
tmpfs            64M     0   64M   0% /dev
shm              64M     0   64M   0% /dev/shm
/dev/root        73G   57G   17G  78% /__w
tmpfs           3.2G  1.2M  3.2G   1% /run/docker.sock
tmpfs           7.9G     0  7.9G   0% /proc/acpi
tmpfs           7.9G     0  7.9G   0% /proc/scsi
tmpfs           7.9G     0  7.9G   0% /sys/firmware
```



after:
```
Filesystem      Size  Used Avail Use% Mounted on
overlay          73G   55G   19G  76% /
tmpfs            64M     0   64M   0% /dev
shm              64M     0   64M   0% /dev/shm
/dev/root        73G   55G   19G  76% /__w
tmpfs           3.2G  1.2M  3.2G   1% /run/docker.sock
tmpfs           7.9G     0  7.9G   0% /proc/acpi
tmpfs           7.9G     0  7.9G   0% /proc/scsi
tmpfs           7.9G     0  7.9G   0% /sys/firmware
Removing large packages
Filesystem      Size  Used Avail Use% Mounted on
overlay          73G   49G   25G  67% /
tmpfs            64M     0   64M   0% /dev
shm              64M     0   64M   0% /dev/shm
/dev/root        73G   49G   25G  67% /__w
tmpfs           3.2G  1.2M  3.2G   1% /run/docker.sock
tmpfs           7.9G     0  7.9G   0% /proc/acpi
tmpfs           7.9G     0  7.9G   0% /proc/scsi
tmpfs           7.9G     0  7.9G   0% /sys/firmware
```
 
So applying a separate image can save ~8G for python tests.

File: dev/spark-test-image/python-310/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    libpython3-dev \
+    libssl-dev \
+    libtiff5-dev \
+    libxml2-dev \
+    openjdk-17-jdk-headless \
+    pkg-config \
+    python3.10 \
+    python3-psutil \
+    qpdf \
+    tzdata \
```
Comment (zhengruifeng) [inline dev/spark-test-image/python-310/Dockerfile:58]: `tzdata` is required in several timezone related tests, in the old image it was installed by `pip install python3.9` as a dependency.

**Review by LuciferYang — APPROVED** [review]
seems fine to me

**Comment by zhengruifeng** [comment]
thanks, merged to master, I will monitor the CI status

**Review by dongjoon-hyun — COMMENTED** [review]
+1, late LGTM.

**Comment by panbingkun** [comment]
+1, late LGTM.

**Pull request closed without merging**